mountpoint-s3-client/src/object_client.rs

```diff
@@ -210,11 +210,28 @@ pub enum ListObjectsError {
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct HeadObjectResult {
-    /// The name of the bcuket
-    pub bucket: String,
+    /// Size of the object in bytes.
+    ///
+    /// Refers to the `Content-Length` HTTP header for HeadObject.
+    pub size: u64,
+
+    /// The time this object was last modified.
+    pub last_modified: OffsetDateTime,
+
+    /// Entity tag of this object.
+    pub etag: String,
```

Comment:
Should we use `ETag` here?

Reply:
We could. I didn't want to diverge from what we have in `ObjectInfo`, but maybe we should go ahead and update both.

I'll update just `HeadObjectResult` for now, and can follow up with a PR for `ObjectInfo` (for `list_objects`).